database/test_database.py

```diff
@@ -0,0 +1,103 @@
+from unittest import TestCase
+
+from database.image import *
+from database.project import *
+
+
+# Before running make sure no .db files are present in execution directory
+
+
+# Tests for Project Class
+class TestDatabase(TestCase):
+    # should not actually be called
+    # the test for insert
+    # insert two rows and verify with fetch
+    # also verify for row which is not there (testing fetch also)
+    def insert_project_test(self):
+        # insert a project
+        pr = ProjectRepository()
+        pr.insert("project 1", "network 1")
+
+        # insert another project
+        pr.insert("project 2", "network 2")
+
+        # check whether first project is present
+        pid = pr.fetch_id_with_name("project 1")
+        self.assertIsNotNone(pid)
+        self.assertEqual(pid, 1)
+
+        # check whether second project is present
+        pid = pr.fetch_id_with_name("project 2")
+        self.assertIsNotNone(pid)
+        self.assertEqual(pid, 2)
+
+        # verify that None is being returned for a project which doesnt exist
+        pid = pr.fetch_id_with_name("project 3")
+        self.assertIsNone(pid)
+
+    # should not actually be called
+    # the test for delete
+    # delete the inserted row
+    # check if it is gone
+    def delete_with_name_project_test(self):
+        # delete first project
+        pr = ProjectRepository()
+        pr.delete_with_name("project 1")
+
+        # verify that it is gone
+        pid = pr.fetch_id_with_name("project 1")
+        self.assertIsNone(pid)
+
+    # should not be called
+    # runs the project test case
+    def project_test(self):
+        self.insert_project_test()
+        self.delete_with_name_project_test()
```

Comment:
You don't need to call the test methods. The test framework will automatically call all methods that start with test. So please rename all the test methods to start with test. 

https://docs.python.org/2.7/library/unittest.html